### ISSUE TITLE
Fix false positives for '*' and '/' operators in function-calc-no-unspaced-operator

### ISSUE THREAD

**Issue by yassin-kammoun-sonarsource** [stylelint/stylelint#5695]
### What steps are needed to reproduce the bug?

```css
a {
  width: calc(1*2);
}
```

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "function-calc-no-unspaced-operator": true
  }
}
```

### How did you run Stylelint?

https://stylelint.io/demo

### Which version of Stylelint are you using?

14.0.1

### What did you expect to happen?

No issues should be raised because [W3C CSS Values and Units Module Level 3 section 8.1.1 Syntax](https://www.w3.org/TR/css-values-3/#calc-syntax) says the following:

> In addition, white space is required on both sides of the + and - operators. (The * and / operaters can be used without white space around them.)

### What actually happened?

The following problems were reported:
- Expected single space before "*" operator (function-calc-no-unspaced-operator)
- Expected single space after "*" operator (function-calc-no-unspaced-operator)

### Does the bug relate to non-standard syntax?

No.

### Proposal to fix the bug

At first glance, the rule implementation should no longer consider `*` and `/` operators. In other words, one would just need to remove these operators from:

https://github.com/stylelint/stylelint/blob/e4a2e13c63867c47a4e48cff4fb0d62cc703765b/lib/rules/function-calc-no-unspaced-operator/index.js#L20

**Comment by ybiquitous** [comment]
@yassin-kammoun-sonarsource Thanks for reporting the issue and using the issue template!

We can reproduce it on the [demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdilB3AllAC4AW8KYSANmABQCMAVAEwCUA3FgL5YgA0IAM4BPTESQAPBOEGD+4CJgBm+AObSM2FOhAAnAK5UYgneU04cOpfsxgi+RQFpKNR5giObggA5IwMKEcIbxhdJCIIXVMUIgMYLBweTCSQLiA). 👍🏼

**Comment by jeddy3** [comment]
As the issue is labelled as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by stof** [comment]
`function-calc-no-unspaced-operator` asks to have no unspaced operators (according to its name), so the existing implementations makes sense to me. Maybe it needs an option `always` vs `when_required_by_spec` instead.

**Comment by jeddy3** [comment]
I suggest this change is accompanied by a new rule: `function-calc-whitespace-around-operator: "always"|"never"`.

`function-calc-no-unspace-operator` would remain in the ["possible errors"](https://stylelint.io/user-guide/rules/list#possible-errors) category, whereas this new rule would [be stylistic](https://stylelint.io/user-guide/rules/list#stylistic-issues).

(Note further stylistic rules [are up for discussion](https://github.com/stylelint/stylelint/issues/5674).)

**Comment by Mouvedia** [comment]
This rule should be renamed.
There are 20 more [mathematical expressions](https://www.w3.org/TR/css-values-4/#math) that accepts ["calc-sum"](https://www.w3.org/TR/css-values-4/#calc-syntax).

I propose `function-math-whitespace: true|false`.

**Comment by jeddy3** [comment]
@Mouvedia Please open a new issue so that this issue stays focused on the about the `calc(1*2)` and `calc(1/2)` false positives. It'll be best if we can discuss how best to tackle the `round(2+2)` false negatives in a new issue.

**Comment by Mouvedia** [comment]
> It'll be best if we can discuss how best to tackle the round(2+2) false negatives in a new issue.

Since, IMO, allowing `calc(1*2)` and `calc(1/2)` would require renaming the rule so that it wouldn't create confusion, I thought that it would be better to tackle both problems at once.
I am not interested in fixing just one issue since both of them require a renaming anyway.
I am against renaming it twice or having a separate rule (while deprecating this one).

**Comment by jeddy3** [comment]
@Mouvedia We try to avoid breaking changes and major releases as they have proven painful for the community, especially in regards to plugins and shared configs. We may be able to address both issues within a minor release by keeping the rule name the same.

We can:

- fix the false positives for `calc(1*2)` and `calc(1/2)` by adding an extended description to the rule that explains that the `+` and `-` operators need to spaced. This works because the rule exists within the [avoid errors category](https://stylelint.io/user-guide/rules/list#avoid-errors) rather than the [stylistic issues one](https://stylelint.io/user-guide/rules/list#stylistic-issues) (which are now frozen). The current name will still make sense in this context.
- fix the false negatives for `round` and other `calc-sum` functions by broadening the meaning of calc to calc-sum, and adjusting the rule description accordingly. As it's fixing false negatives, it'd be a minor release, in line with [our semantic-versioning policy](https://stylelint.io/about/semantic-versioning#minor-release).

I'm confident about the first, but less so about the second. However, it may be the best path forward that benefits the majority of our users.
